fury/ui/helpers.py

```diff
@@ -23,6 +22,81 @@ def clip_overflow(textblock, width, side='right'):
     -------
     clipped text : str
         Clipped version of the text.
+    """
+    original_str = textblock.message
+    prev_bg = textblock.have_bg
+
+    clip_idx = check_overflow(textblock, width, '...', side)
+
+    if clip_idx == 0:
+        return original_str
+
+    textblock.have_bg = prev_bg
+    return textblock.message
+
+
+def wrap_overflow(textblock, wrap_width, side='right'):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+    side : str, optional
+        Clips the overflowing text according to side.
+        It takes values "left" or "right".
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    str_copy = textblock.message
+    prev_bg = textblock.have_bg
+    wrap_idxs = []
+
+    wrap_idx = check_overflow(textblock, wrap_width, '', side)
+
+    if wrap_idx == 0:
+        return original_str
+
+    wrap_idxs.append(wrap_idx)
+
+    while wrap_idx != 0:
+        str_copy = str_copy[wrap_idx:]
+        textblock.message = str_copy
+        wrap_idx = check_overflow(textblock, wrap_width, '', side)
+        if wrap_idx != 0:
+            wrap_idxs.append(wrap_idxs[-1]+wrap_idx+1)
+
+    for idx in wrap_idxs:
+        original_str = original_str[:idx] + '\n' + original_str[idx:]
+
+    textblock.message = original_str
+    textblock.have_bg = prev_bg
+    return textblock.message
+
+
+def check_overflow(textblock, width, overflow_postfix='',
+                   side='right'):
+    """Checks if the text is overflowing.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text is to be checked.
+    width: int
+        Required width of the text.
+    overflow_postfix: str, optional
+        Postfix to be added to the text if it is overflowing.
+
+    Returns
+    -------
+    int
+
```

Comment:
You need to explain what this integer represents. @antrikshmisri

Reply:
Done.